net/netcheck/netcheck.go

```diff
@@ -980,6 +1015,38 @@ func (c *Client) finishAndStoreReport(rs *reportState, dm *tailcfg.DERPMap) *Rep
 	return report
 }
 
+const generate204URL = "http://controlplane.tailscale.com/generate_204"
+
+var noRedirectClient = &http.Client{
+	// No redirects allowed
+	CheckRedirect: func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	},
+
+	// Remaining fields are the same as the default client.
+	Transport: http.DefaultClient.Transport,
+	Jar:       http.DefaultClient.Jar,
+	Timeout:   http.DefaultClient.Timeout,
+}
+
+// checkCaptivePortal returns whether or not we think the system is behind a
+// captive portal, detected by making a request to a URL that we know should
+// return a "204 No Content" response and checking if that's what we get.
+//
+// The boolean return is whether we think we have a captive portal.
+func (c *Client) checkCaptivePortal(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, captivePortalTimeout)
+	defer cancel()
+
+	req, _ := http.NewRequestWithContext(ctx, "GET", generate204URL, nil)
```

Comment:
Let's not throw away the error here, if someone changes this code in the future this might make the Do call panic.